datacube/utils/dask.py

```diff
@@ -0,0 +1,270 @@
+                    **kw)
+
+    return client
+
+
+def _randomize(prefix):
+    return '{}-{:08x}'.format(prefix, randint(0, 0xFFFFFFFF))
+
+
+def partition_map(n: int, func: Any, its: Iterable[Any],
+                  name: str = 'compute') -> Iterable[Any]:
+    """ Partition sequence into lumps of size `n`, then construct dask delayed computation evaluating to:
+
+    [func(x) for x in its[0:1n]],
+    [func(x) for x in its[n:2n]],
+    ...
+    [func(x) for x in its[]],
+
+    :param n: number of elements to process in one go
+    :param func: Function to apply (non-dask)
+    :param its:  Values to feed to fun
+    :param name: How the computation should be named in dask visualizations
+    """
+    def lump_proc(dd):
+        return [func(d) for d in dd]
+
+    proc = dask.delayed(lump_proc, nout=1, pure=True)
+    data_name = _randomize('data_' + name)
+    name = _randomize(name)
+
+    for i, dd in enumerate(toolz.partition_all(n, its)):
+        lump = dask.delayed(dd,
+                            pure=True,
+                            traverse=False,
+                            name=data_name + str(i))
+        yield proc(lump, dask_key_name=name + str(i))
+
+
+def compute_tasks(tasks: Iterable[Any], client: Client,
+                  max_in_flight: int = 3) -> Iterable[Any]:
+    """ Parallel compute stream with back pressure.
+
+        Equivalent to:
+
+        (client.compute(task).result()
+          for task in tasks)
+
+        but with up to `max_in_flight` tasks being processed at the same time.
+        Input/Output order is preserved, so there is a possibility of head of
+        line blocking.
+
+        NOTE: lower limit is 3 concurrent tasks to simplify implementation,
+              there is no point calling this function if you want one active
+              task and supporting exactly 2 active tasks is not worth the complexity,
+              for now. We might special-case `2` at some point.
+
+    """
+    # New thread:
+    #    1. Take dask task from iterator
+    #    2. Submit to client for processing
+    #    3. Send it of to wrk_q
+    #
+    # Calling thread:
+    #    1. Pull scheduled future from wrk_q
+    #    2. Wait for result of the future
+    #    3. yield result to calling code
+    from .generic import it2q, qmap
+
+    # (max_in_flight - 2) -- one on each side of queue
+    wrk_q = queue.Queue(maxsize=max(1, max_in_flight - 2))  # type: queue.Queue
+
+    # fifo_timeout='0ms' ensures that priority of later tasks is lower
+    futures = (client.compute(task, fifo_timeout='0ms') for task in tasks)
+
+    in_thread = threading.Thread(target=it2q, args=(futures, wrk_q))
+    in_thread.start()
+
+    yield from qmap(lambda f: f.result(), wrk_q)
+
+    in_thread.join()
+
+
+def pmap(func: Any,
+         its: Iterable[Any],
+         client: Client,
+         lump: int = 1,
+         max_in_flight: int = 3,
+         name: str = 'compute') -> Iterable[Any]:
+    """ Parallel map with back pressure.
```

Comment:
Better name should probably refer to back pressure somehow, since this is in dask context and parallel is kinda implied, like dask version of parallel map is just `map`.